forte/data/data_store.py

```diff
@@ -433,10 +434,30 @@ def get_entry(self, tid: int) -> Tuple[List, int, int]:
             The entry which `tid` corresponds to, its `type_id` and its index
             in the `type_name` list.
         """
+        if tid not in self.__entry_dict:
+            raise ValueError(f"Entry with tid {tid} not found.")
+        entry = self.__entry_dict[tid]
+        entry_type = entry[self.__entry_type_idx]
+        if entry_type not in self.__type_index_dict:
+            raise KeyError(f"Entry of type {entry_type} is not found.")
+        type_id = self.__type_index_dict[entry_type]
+        index_id = -1
+
         # If the entry is an annotation, bisect the annotation sortedlist
         # to find the entry. May use LRU cache to optimize speed.
         # Otherwise, use `index_id` to find the index of the entry.
-        raise NotImplementedError
+        if issubclass(get_class(entry_type), (Annotation, AudioAnnotation)):
```

Comment:
`issubclass` check and `get_class` might slow down the process. @wanglec @qinzzz do we have the class hierarchy somewhere?

Another idea is that we should cache the subclass information (e.g. using an `is_annotation_dict`) and only check it the first time.

Reply:
#628 provides a [_is_annotation()](https://github.com/asyml/forte/pull/628/files#diff-f1c68405498aa1cba27f81862f28d3edecfdc2d8501394bb3cac9d83dfa1283dR250) method and this line will be updated when #628 is merged.